Sources/Sentry/SentryTracer.m

```diff
@@ -109,7 +109,12 @@ - (void)captureTransaction
 {
     NSArray *spans;
     @synchronized(_spans) {
-        spans = [_spans copy];
+        spans = [_spans
+            filteredArrayUsingPredicate:[NSPredicate
+                                            predicateWithBlock:^BOOL(id<SentrySpan> _Nullable span,
+                                                NSDictionary<NSString *, id> *_Nullable bindings) {
+                                                return span.isFinished;
+                                            }]];
```

Comment:
`h`: I think we have to do this in the client as someone could manually call `[hub captureEvent:transaction]`.

Reply:
No, the user doesn't have a reference to transaction. All it have is a span that cannot be used on captureEvent.